plugins/default/src/Ide/Plugin/Eval.hs

```diff
@@ -280,38 +277,71 @@ runEvalCmd lsp state EvalParams {..} = withIndefiniteProgress lsp "Eval" Cancell
 
     return (WorkspaceApplyEdit, ApplyWorkspaceEditParams workspaceEdits)
 
+
+type GHCiLikeCmd = DynFlags -> Text -> Ghc (Maybe Text)
+
+-- Should we use some sort of trie here?
+ghciLikeCommands :: [(Text, GHCiLikeCmd)]
+ghciLikeCommands =
+  [ ("kind", doKindCmd False)
+  , ("kind!", doKindCmd True)
+  , ("type", doTypeCmd)
+  ]
+
 evalGhciLikeCmd :: Text -> Text -> Ghc (Maybe Text)
 evalGhciLikeCmd cmd arg = do
   df <- getSessionDynFlags
-  let tppr = T.pack . showSDoc df . ppr
-  case cmd of
-    "kind" -> do
-      let input = T.strip arg
-      (_, kind) <- typeKind False $ T.unpack input
-      pure $ Just $ "-- " <> input <> " :: " <> tppr kind <> "\n"
-    "kind!" -> do
-      let input = T.strip arg
-      (ty, kind) <- typeKind True $ T.unpack input
-      pure
-        $ Just
-        $ T.unlines
-        $ map ("-- " <>)
-        [ input <> " :: " <> tppr kind
-        , "= " <> tppr ty
-        ]
-    "type" -> do
-      let (emod, expr) = parseExprMode arg
-      ty <- exprType emod $ T.unpack expr
-      pure $ Just $
-        "-- " <> expr <> " :: " <> tppr ty <> "\n"
-    _ -> E.throw $ GhciLikeCmdNotImplemented cmd arg
+  case lookup cmd ghciLikeCommands
+    <|> snd <$> find (T.isPrefixOf cmd . fst) ghciLikeCommands of
+    Just hndler -> hndler df arg
+    _           -> E.throw $ GhciLikeCmdNotImplemented cmd arg
+
+doKindCmd :: Bool -> DynFlags -> Text -> Ghc (Maybe Text)
+doKindCmd False df arg = do
+  let input = T.strip arg
+  (_, kind) <- typeKind False $ T.unpack input
+  let kindText = text (T.unpack input) <+> "::" <+> ppr kind
+  pure
+    $ Just
+    $ T.unlines
+    $ map ("-- " <>)
+    $ T.lines (T.pack (showSDoc df kindText))
+doKindCmd True df arg = do
+  let input = T.strip arg
+  (ty, kind) <- typeKind True $ T.unpack input
+  let kindDoc = text (T.unpack input) <+> "::" <+> ppr kind
+      tyDoc = "=" <+> ppr ty
+  pure
+    $ Just
+    $ T.unlines
+    $ map ("-- " <>)
+    $ T.lines
```

Comment:
This logic to format the output of evaluation is repeated in several places and easy to get wrong (as this PR shows). 
Have you considered extracting it into a helper function or event better, into `evalGhciLikeCmd` ?

Reply:
Good catch! I've just pushed the commit that removes commenting-out works from each `do*Cmd` and integrates it to `evalGhciLikeCmd`.